plugins/module_utils/redfish_utils.py

```diff
@@ -3520,3 +3520,119 @@ def delete_volumes(self, storage_subsystem_id, volume_ids):
 
         return {'ret': True, 'changed': True,
                 'msg': "The following volumes were deleted: %s" % str(volume_ids)}
+
+    def get_bios_registry(self):
+        # Get /redfish/v1
+        response = self.get_request(self.root_uri + self.service_root)
+        if not response["ret"]:
+            # error_msg(module, "GET", base_uri, redfish_data.status, redfish_data.text)
+            return response
+
+        server_details = response["data"]
+
+        # Get Registries URI
+        if "Registries" not in server_details:
+            msg = "Getting Registries URI failed, Key 'Registries' not found in /redfish/v1/ response: %s"
+            # module.fail_json(msg=msg % str(server_details))
+            return {
+                "ret": False,
+                "msg": msg % str(server_details)
+            }
+
+        reg_uri = server_details["Registries"]["@odata.id"]
+        reg_resp = self.get_request(self.root_uri + reg_uri)
+        if not reg_resp["ret"]:
+            # error_msg(module, "GET", reg_uri, reg_resp.status, reg_resp.text)
+            return reg_resp
+
+        reg_data = reg_resp["data"]
+
+        # Get BIOS attribute registry URI
+        lst = []
+        response = self.check_attribute_registry_uri(reg_data, reg_uri)
+        if not response["ret"]:
+            return response
+
+        resp_data, resp_uri = response["resp_data"], response["resp_uri"]
+        # Get the location URI
+        response = self.check_location_uri(resp_data, resp_uri)
+        if not response["ret"]:
+            return response
+
+        rsp_data, rsp_uri = response["rsp_data"], response["rsp_uri"]
+
+        if "RegistryEntries" not in rsp_data:
+            return {
+                "msg": "'RegistryEntries' not present in %s response, %s" % (rsp_uri, str(rsp_data)),
+                "ret": False
+            }
+
+        return {
+            "bios_registry": rsp_data,
+            "bios_registry_uri": rsp_uri,
+            "ret": True
+        }
+
+    def check_attribute_registry_uri(self, reg_data, reg_uri):
+        # Get BiosAttributeRegistry<BIOS version> response
```

Comment:
The logic in here doesn't seem correct at all. There's no requirement for the name of the BIOS attribute registry to contain the term "attribute". The location of the appropriate MessageRegistryFile resource is determined like this:

* Perform a GET operation on the Bios resource of the desired system.
* Look at the value of the AttributeRegistry property
* Find the MessageRegistryFile resource be either...
    * Append the value of the AttributeRegistry property from the previous step to /redfish/v1/Registries/
    * Iterate over the MessageRegistryFileCollection and find the resource whose Id property matches AttributeRegistry

Reply:
Agree, this function has been modified with below procedure,

Perform a GET operation on the Bios resource of the desired system.
Look at the value of the AttributeRegistry property
Find the MessageRegistryFile resource be either...
        Append the value of the AttributeRegistry property from the previous step to /redfish/v1/Registries/
        Iterate over the MessageRegistryFileCollection and find the resource whose Id property matches AttributeRegistry